File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_SEGA_LEONEL_ISAIAS/Introduccion/2.py
class Bus:
    def __init__(self, capacidad):
        self.capacidad = capacidad
        self.pasajeros = 0
        self.costo_pasaje = 1.50

    def subir_pasajeros(self, cantidad):
        if self.pasajeros + cantidad <= self.capacidad:
            self.pasajeros += cantidad
        else:
            print("⚠️ No hay suficientes asientos disponibles.")

    def cobrar_pasaje(self):
        total = self.pasajeros * self.costo_pasaje
        print(f"Total cobrado: Bs. {total:.2f}")

    def asientos_disponibles(self):
        print(f"Asientos disponibles: {self.capacidad - self.pasajeros}")

# Ejemplo
bus = Bus(30)
bus.subir_pasajeros(20)
bus.cobrar_pasaje()
bus.asientos_disponibles()

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_SEGA_LEONEL_ISAIAS/Introduccion/2.java
public class Bus {
    private int capacidad;
    private int pasajeros;
    private double costoPasaje;

    public Bus(int capacidad, double costoPasaje) {
        this.capacidad = capacidad;
        this.costoPasaje = costoPasaje;
        this.pasajeros = 0;
    }

    public void subirPasajeros(int cantidad) {
        if (pasajeros + cantidad <= capacidad) {
            pasajeros += cantidad;
        } else {
            System.out.println("No hay suficientes asientos disponibles.");
        }
    }

    public void cobrarPasaje() {
        double total = pasajeros * costoPasaje;
        System.out.println("Total cobrado: Bs. " + total);
    }

    public void mostrar() {
        System.out.println("Capacidad: " + capacidad + 
                           " | Pasajeros: " + pasajeros + 
                           " | Costo pasaje: " + costoPasaje);
    }

    public static void main(String[] args) {
        Bus b1 = new Bus(30, 2.50);
        b1.subirPasajeros(20);
        b1.mostrar();
        b1.cobrarPasaje();
    }
}

